### PR TITLE
Add several functions to SimulatorAccess.

### PR DESCRIPTION
**Pull request by bangerth**
This just adds a few functions to SimulatorAccess that I needed for separate work but that might also be useful elsewhere -- so it put them into a pull request of their own.

### CODE REVIEW

**Comment by tjhei** [comment]
the compiler didn't like this:
```
/aspect-tester/aspect/source/simulator/simulator_access.cc:311:53: error: no member named 'get' in 'std::map<unsigned char, std::shared_ptr<aspect::TractionBoundaryConditions::Interface<2> >, std::less<unsigned char>, std::allocator<std::pair<const unsigned char, std::shared_ptr<aspect::TractionBoundaryConditions::Interface<2> > > > >'
    Assert (simulator->traction_boundary_conditions.get() != 0,
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
```

**Comment by bangerth** [comment]
I fixed the miscompile. Sorry for that, it was drowned in the noise of the other miscompilations of #695

File: include/aspect/simulator_access.h [hunk 1]
```diff
@@ -379,13 +398,20 @@ namespace aspect
...
       /**
-       * Return a pointer to the object that describes the temperature
+       * Return a reference to the object that describes the temperature
        * boundary values.
        */
       const BoundaryTemperature::Interface<dim> &
       get_boundary_temperature () const;
 
       /**
+       * Return a reference to the object that traction temperature
```
Comment (gassmoeller) [inline include/aspect/simulator_access.h]: There is a copy-paste error in this sentence.

**Comment by gassmoeller** [comment]
Looks good apart from this one sentence in the documentation. Please fix and merge yourself.

File: include/aspect/simulator_access.h (hunk 1 above)
Comment (bangerth) [inline include/aspect/simulator_access.h]: Good point. Fixed.

**Pull request merged**